pkg/protocol/engine/booker/inmemorybooker/booker.go

```diff
@@ -158,9 +170,12 @@ func (b *Booker) inheritConflicts(block *blocks.Block) (conflictIDs ds.Set[iotag
 		case iotago.WeakParentType:
 			conflictIDsToInherit.AddAll(parentBlock.PayloadConflictIDs())
 		case iotago.ShallowLikeParentType:
-			// TODO: check whether it contains a (conflicting) TX, otherwise shallow like reference is invalid?
-			//  if a block contains a transaction that itself is not conflicting, then it's possible to vote on any transaction in the UTXO-future cone of the conflict
-			//  NOTE: the above only applies when we don't fork all transactions.
+			// Check whether the parent contains a conflicting TX,
+			// otherwise reference is invalid and the block should be marked as invalid as well.
+			if tx, hasTx := parentBlock.Transaction(); !hasTx || !parentBlock.PayloadConflictIDs().Has(lo.PanicOnErr(tx.ID(b.apiProvider.APIForSlot(block.ID().Index())))) {
```

Comment:
Shouldn't you load the API for the parentBlock's index rather than block's index?